services/management/commands/turku_service_import/addresses.py

```diff
@@ -0,0 +1,95 @@
+import csv
+import os
+
+from django.conf import settings
+from django.contrib.gis.geos import Point
+from munigeo.models import Address, Municipality, Street
+
+
+class AddressImporter:
+    def __init__(self, logger):
+        self.logger = logger
+
+        if hasattr(settings, 'PROJECT_ROOT'):
+            root_dir = settings.PROJECT_ROOT
+        else:
+            root_dir = settings.BASE_DIR
+        self.data_path = os.path.join(root_dir, 'data')
+
+        self.csv_field_names = ('city', 'street', 'street_number', 'longitude', 'latitude')
+        self.valid_cities = ['turku', 'åbo']
+
+    def _import_address(self, entry):
+        street, _ = Street.objects.get_or_create(**entry['street'])
+        location = Point(
+            srid=3877,
+            **entry['point']
+        )
+
+        address, created = Address.objects.get_or_create(
+            street=street,
+            defaults={'location': location},
+            **entry['address']
+        )
+
+        if created:
+            address.refresh_from_db()
+
+    def _create_address_mapping(self, address_reader):
+        turku = Municipality.objects.get(id='turku')
+        multi_lingual_addresses = {}
+        for row in address_reader:
+            if row['city'].lower() not in self.valid_cities:  # Turku only
+                continue
+
+            coordinates = row['longitude'] + row['latitude']
+            if coordinates not in multi_lingual_addresses:
+                multi_lingual_addresses[coordinates] = {
+                    'street': {
+                        'municipality': turku,
+                    },
+                    'point': {
+                        'x': float(row['longitude']),
+                        'y': float(row['latitude']),
+                    },
+                    'address': {
+                        'number': row['street_number'],
+                    }
+                }
+
+            if row['city'].lower() == 'turku':
+                multi_lingual_addresses[coordinates]['street']['name'] = row['street']
+                multi_lingual_addresses[coordinates]['street']['name_en'] = row['street']
+            elif row['city'].lower() == 'åbo':
+                # As we expect Swedish names to come after Finnish ones, we always default
+                # to the Swedish name, but if there are a Swedish only street name, we
+                # fall back to Swedish for all name entries
+                if 'name' not in multi_lingual_addresses[coordinates]['street']:
+                    multi_lingual_addresses[coordinates]['street']['name'] = row['street']
+                    multi_lingual_addresses[coordinates]['street']['name_en'] = row['street']
```

Comment:
Can we actually count on that Swedish names will always come after Finnish ones? And what if there is no Swedish translation at all, shouldn't we use Finnish as a fallback then?

To address these issues, maybe first go through all the rows and add all the languages we get, and then as a separate step populate all missing languages by some fallback strategy?

Also, now looking at HKI Palvelukartta backend, it doesn't seem to have English street names at all, so those aren't probably needed either.

Reply:
Regarding the English names: I will remove them

Regarding the swe/fin stuff:
The comment is bad here, I will rewrite, the code does what you said. It will always use Finnish if available, otherwise it will use Swedish as the `name`. Since I will be use `name_fi` instead of name, I will also place the Swedish name as the Finnish one (if there is no Finnish name) since I'm guessing that not having a Finnish name might break stuff at some point in the code since I'm not super familiar with how the addresses are used. This is also really easy to change at a later stage.